rioxarray/_io.py

```diff
@@ -0,0 +1,392 @@
+        ``transform`` attribute or not. The default is to automatically
+        parse the coordinates only if they are rectilinear (1D).
+        It can be useful to set ``parse_coordinates=False``
+        if your files are very large or if you don't need the coordinates.
+    chunks : int, tuple or dict, optional
+        Chunk sizes along each dimension, e.g., ``5``, ``(5, 5)`` or
+        ``{'x': 5, 'y': 5}``. If chunks is provided, it used to load the new
+        DataArray into a dask array. Chunks can also be set to
+        ``True`` or ``"auto"`` to choose sensible chunk sizes according to
+        ``dask.config.get("array.chunk-size").
+    cache : bool, optional
+        If True, cache data loaded from the underlying datastore in memory as
+        NumPy arrays when accessed to avoid reading from the underlying data-
+        store multiple times. Defaults to True unless you specify the `chunks`
+        argument to use dask, in which case it defaults to False.
+    lock : False, True or threading.Lock, optional
+        If chunks is provided, this argument is passed on to
+        :py:func:`dask.array.from_array`. By default, a global lock is
+        used to avoid issues with concurrent access to the same file when using
+        dask's multithreaded backend.
+    masked : bool, optional
+        If True, read the mask and to set values to NaN. Defaults to False.
+return
+    Returns
+    -------
+    data : DataArray
+        The newly created DataArray.
+    """
+    import rasterio
+    from rasterio.vrt import WarpedVRT
+
+    vrt_params = None
+    if isinstance(filename, rasterio.io.DatasetReader):
+        filename = filename.name
+    elif isinstance(filename, rasterio.vrt.WarpedVRT):
+        vrt = filename
+        filename = vrt.src_dataset.name
+        vrt_params = dict(
+            crs=vrt.crs.to_string(),
+            resampling=vrt.resampling,
+            src_nodata=vrt.src_nodata,
+            dst_nodata=vrt.dst_nodata,
+            tolerance=vrt.tolerance,
+            transform=vrt.transform,
+            width=vrt.width,
+            height=vrt.height,
+            warp_extras=vrt.warp_extras,
+        )
+
+    if lock is None:
+        lock = RASTERIO_LOCK
+
+    manager = CachingFileManager(rasterio.open, filename, lock=lock, mode="r")
+    riods = manager.acquire()
+    if vrt_params is not None:
+        riods = WarpedVRT(riods, **vrt_params)
+
+    if cache is None:
+        cache = chunks is None
+
+    coords = OrderedDict()
+
+    # Get bands
+    if riods.count < 1:
+        raise ValueError("Unknown dims")
+    coords["band"] = np.asarray(riods.indexes)
+
+    # Get coordinates
+    if LooseVersion(rasterio.__version__) < "1.0":
+        transform = riods.affine
+    else:
+        transform = riods.transform
+    if transform.is_rectilinear:
+        # 1d coordinates
+        parse = True if parse_coordinates is None else parse_coordinates
```

Comment:
Is `parse` supposed to be used somewhere other than the `if` statement below? If not maybe just fold the assignment logic into the `if` statement. Same below for 2d coordinates.

Reply:
Good catch, I agree.